parsely/parselyandroid/ParselyTracker.java

```diff
@@ -16,177 +16,442 @@
-        if (queue == null || queue.size() == 0) {
+    /*! \brief Returns whether the engagement tracker is running.
+     *
+     * @return Whether the engagement tracker is running.
+     */
+    public boolean engagementIsActive() {
+        return this.engagementManager != null;
+    }
+
+    /*! \brief Returns whether video tracking is active.
+     *
+     * @return Whether video tracking is active.
+     */
+    public boolean videoIsActive() {
+        return this.videoEngagementManager != null;
+    }
+
+    /*! \brief Returns the interval at which the event queue is flushed to Parse.ly.
+     *
+     * @return The interval at which the event queue is flushed to Parse.ly.
+     */
+    public long getFlushInterval() {
+        return this.flushManager.getIntervalMillis() / 1000;
+    }
+
+    /*! \brief Getter for this.isDebug
+     *
+     * @return Whether debug mode is active.
+     */
+    public boolean getDebug() {
+        return isDebug;
+    }
+
+    /*! \brief Set a debug flag which will prevent data from being sent to Parse.ly
+     *
+     *  Use this flag when developing to prevent the SDK from actually sending requests
+     *  to Parse.ly servers. The value it would otherwise send is logged to the console.
+     *
+     *  @param debug Value to use for debug flag.
+     */
+    public void setDebug(boolean debug) {
+        isDebug = debug;
+        PLog("Debugging is now set to " + isDebug);
+    }
+
+    /*! \brief Register a pageview event using a URL and optional metadata.
+     *
+     *  @param url         The canonical URL of the article being tracked
+     *                     (eg: "http://example.com/some-old/article.html")
+     *  @param urlMetadata Optional metadata for the URL -- not used in most cases. Only needed
+     *                     when `url` isn't accessible over the Internet (i.e. app-only
+     *                     content). Do not use for URLs that Parse.ly would normally crawl.
+     *  @param extraData   A Map of additional information to send with the event.
+     */
+    public void trackPageview(
+            @NonNull String url,
+            @Nullable String urlRef,
+            @Nullable ParselyMetadata urlMetadata,
+            @Nullable Map<String, Object> extraData) {
+        if (url == null || url.equals("")) {
+            throw new NullPointerException("url cannot be null or empty.");
+        }
+        this.enqueueEvent(this.buildEvent(url, urlRef, "pageview", urlMetadata, extraData));
+    }
+
+    /*! \brief Start engaged time tracking for the given URL.
+     *
+     * This starts a timer which will send events to Parse.ly on a regular basis
+     * to capture engaged time for this URL. The value of `url` should be a URL for
+     * which `trackPageview` has been called.
+     *
+     * @param url The URL to track engaged time for.
+     */
+    public void startEngagement(@NonNull String url, @Nullable String urlRef) {
```

Comment:
The equivalent function in the iOS SDK accepts `extraData`

Reply:
Same argument as before. We don't send this or metadata in heartbeat events, nor do I think we should. We don't support it right now, and if someday we have a need to support `extraData` in heartbeat events, we can add it to a `3.1.0` release.